testsuite/tests/typing-jkind-bounds/subsumption/modalities.ml

```diff
@@ -0,0 +1,320 @@
+|}]
+
+module M : sig
+  type 'a t : value mod unique
+end = struct
+  type 'a t : immediate with 'a @@ aliased
+end
+[%%expect {|
+module M : sig type 'a t : value mod unique end
+|}]
+
+module M : sig
+  type 'a t : value mod global unique many portable uncontended
+end = struct
+  type 'a t : immediate with 'a @@ aliased many contended global portable
+end
+[%%expect {|
+module M : sig type 'a t : immutable_data mod global unique end
+|}]
+
+module M : sig
+  type ('a, 'b) t : value mod portable
+end = struct
+  type ('a, 'b) t : value mod portable with 'a @@ portable with 'b @@ portable
+end
+[%%expect {|
+module M : sig type ('a, 'b) t : value mod portable end
+|}]
+
+type ('a, 'b) t : value mod portable with 'a @@ portable with 'b @@ contended
+
+module type S = sig
+  type ('a, 'b) t : value mod portable
+end
+
+module type T = S with type ('a, 'b) t = ('a, 'b) t
+[%%expect {|
+type ('a, 'b) t : value mod portable with 'b
+module type S = sig type ('a, 'b) t : value mod portable end
+Line 7, characters 23-51:
+7 | module type T = S with type ('a, 'b) t = ('a, 'b) t
+                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "('a, 'b) t" is value mod portable with 'b
+         because of the definition of t at line 1, characters 0-77.
+       But the kind of type "('a, 'b) t" must be a subkind of
+         value mod portable
+         because of the definition of t at line 4, characters 2-38.
+|}]
+
+module M : sig
+  type ('a, 'b) t : value mod portable with 'b
+end = struct
+  type ('a, 'b) t : value mod portable with 'a @@ portable with 'b @@ contended
+end
+[%%expect {|
+module M : sig type ('a, 'b) t : value mod portable with 'b end
+|}]
+
+module M : sig
+  type 'a t : immutable_data with 'a @@ portable
+end = struct
+  type 'a t : immutable_data with 'a @@ portable with 'a
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type 'a t : immutable_data with 'a @@ portable with 'a
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type 'a t : immutable_data with 'a end
+       is not included in
+         sig type 'a t : immutable_data with 'a end
+       Type declarations do not match:
+         type 'a t : immutable_data with 'a
+       is not included in
+         type 'a t : immutable_data with 'a
+       The kind of the first is immutable_data with 'a
+         because of the definition of t at line 4, characters 2-56.
+       But the kind of the first must be a subkind of immutable_data with 'a
+         because of the definition of t at line 2, characters 2-48.
+|}]
+
+module M : sig
+  type 'a t : immutable_data with 'a @@ portable
+end = struct
+  type 'a t : immutable_data with 'a with 'a @@ portable
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type 'a t : immutable_data with 'a with 'a @@ portable
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type 'a t : immutable_data with 'a end
```

Comment:
This is a pretty terrible error message. I think there might be a bug here

Reply:
I think this might be the same as the printing bug where `@@ contended` was dropped.